test/e2e/case19_ns_selector_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"open-cluster-management.io/config-policy-controller/test/utils"
+)
+
+const (
+	case19PolicyName  string = "policy-configmap-selector-e2e"
+	case19PolicyYaml  string = "../resources/case19_ns_selector/case19_cm_policy.yaml"
+	case19PrereqYaml  string = "../resources/case19_ns_selector/case19_cm_manifest.yaml"
+	case19PatchPrefix string = "[{\"op\":\"replace\",\"path\":\"/spec/namespaceSelector\",\"value\":"
+	case19PatchSuffix string = "}]"
+)
+
+// Test setup for namespace selection policy tests:
+// - Namespaces `case19-[1-5]-e2e`, each with a `name: <ns-name>` label
+// - Single deployed Configmap `configmap-selector-e2e` in namespace `case19-1-e2e`
+// - Deployed policy should be compliant since it matches the single deployed ConfigMap
+// - Policies are patched so that the namespace doesn't match and should be NonCompliant
+var _ = Describe("Test object namespace selection", func() {
+	// NamespaceSelector patches to test
+	resetPatch := "{\"include\":[\"case19-1-e2e\"]}"
+	patches := []struct {
+		description string
+		patch       string
+		message     string
+	}{
+		{"no namespaceSelector specified", "{}", "namespaced object has no namespace specified"},
+		{
+			"a non-matching LabelSelector",
+			"{\"matchLabels\":{\"name\":\"not-a-namespace\"}}",
+			"namespaced object has no namespace specified",
```

Comment:
This seems like an odd message. I wonder if we need a new message for the case where no namespaces were matched by the selector - that seems different than "not specified". From this message, a user might expect they have to add the namespace to the object template, which is not correct.

Reply:
It shouldn't be necessary to update the message in this PR. I just wanted to flag it while it was more obvious.